Sources/CADCacheAdvance/CADCacheAdvance.swift

```diff
@@ -0,0 +1,124 @@
+    ///   - shouldOverwriteOldMessages: When `true`, once the on-disk store exceeds maximumBytes, new entries will replace the oldest entry.
+    ///
+    /// - Warning: `maximumBytes` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will prevent appending new messages to this cache.
+    /// - Warning: `shouldOverwriteOldMessages` must be consistent for the life of a cache. Changing this value after logs have been persisted to a cache will prevent appending new messages to this cache.
+    @objc
+    public init(
+        fileURL: URL,
+        maximumBytes: Bytes,
+        shouldOverwriteOldMessages: Bool)
+        throws
+    {
+        cache = try CacheAdvance<Storage>(
+            fileURL: fileURL,
+            maximumBytes: maximumBytes,
+            shouldOverwriteOldMessages: shouldOverwriteOldMessages)
+    }
+
+    // MARK: Public
+
+    @objc
+    public var fileURL: URL {
+        cache.fileURL
+    }
+
+    /// Checks if the all the header metadata provided at initialization matches the persisted header. If not, the cache is not writable.
+    /// - Returns: `true` if the cache is writable.
+    @objc
+    public var isWritable: Bool {
+        (try? cache.isWritable()) ?? false
+    }
+
+    /// Appends a message to the cache.
+    /// - Parameter message: A message to write to disk. Must be smaller than both `maximumBytes - FileHeader.expectedEndOfHeaderInFile` and `MessageSpan.max`.
+    @objc
+    public func appendMessage(_ message: String) throws {
+        try cache.append(message: Storage(message))
+    }
+
+    /// Appends a data message to the cache.
+    /// - Parameter message: A message to write to disk. Must be smaller than both `maximumBytes - FileHeader.expectedEndOfHeaderInFile` and `MessageSpan.max`.
+    @objc
+    public func appendData(_ data: Data) throws {
+        try cache.append(message: Storage(data))
+    }
+
+    /// - Returns: `true` when there are no messages written to the file, or when the file can not be read.
+    @objc
+    public var isEmpty: Bool {
+        (try? cache.isEmpty()) ?? true
+    }
+
+    /// Fetches all messages from the cache.
+    @objc
+    public func messageData() throws -> [Data] {
+        try cache.messages().map { $0.d }
+    }
+
+    /// Fetches all messages from the cache.
+    @objc
+    public func messages() throws -> [String] {
+        try cache.messages().map { $0.asString }
+    }
+
+    // MARK: Private
+
+    private let cache: CacheAdvance<Storage>
+}
+
+// MARK: - Storage
+
+private struct Storage: Codable {
+
+    // MARK: Initialization
+
+    fileprivate init(_ string: String) {
+        d = Data(string.utf8)
+    }
+
+    fileprivate init(_ data: Data) {
+        d = data
+    }
+
+    // MARK: Fileprivate
+
+    fileprivate let d: Data
+
+    fileprivate var asString: String {
+        String(decoding: d, as: UTF8.self)
```

Comment:
I think a base64 encoding would be more flexible in the event there are `null` bytes in the data. I think String may act poorly (panic?) if the data provided isn't valid UTF-8.

Reply:
The issue with base64 encoding is that there's is no non-optional api to convert from String to base64 Data. So we'd have to `throw` on failure to encode.

I'm thinking it may make sense to just get rid of the `String` methods entirely, and let folk encode to their own liking. Thoughts @martinmroz?

Reply:
I looked into this once. If it finds invalid data I believe it uses a replacement character.
That being said, I like the idea of removing the string helpers and letting consumers decide what to do.
It's easy to write a category.

Reply:
If we go this path let's simplify the method names and documentation for the data ones that we have now 👍

Reply:
Simplified to just a Data API in https://github.com/dfed/CacheAdvance/pull/42/commits/bd190750a11fd01a582dd969960ccae63d385ebf